### PR TITLE
feat(dom): DOM - toContainElement

### PR DESCRIPTION
**Pull request by matycarolina**
Added the toContainElement matcher.

### CODE REVIEW

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx [hunk 1]
```diff
@@ -42,4 +53,81 @@ describe("[Unit] ElementAssertion.test.ts", () => {
       });
     });
   });
+
+  describe(".toContainElement", () => {
+    context("when the ancestor element contains the descendant element", () => {
+      it("returns the assertion instance", () => {
+        const { getByTestId } = render(<ContainsTestComponent/>);
+
+        const grandparent = getByTestId("grandparent");
```
Comment (fonsiher) [inline packages/dom/test/unit/lib/ElementAssertion.test.tsx:62]: You can use here async/await functions eg: ` const grandparent =  await getByTestId("grandparent");`, just to ensure that asynchronous operations complete before assertions, that also prevents false positive/negatives due unresolved promises.

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx [hunk 2]
```diff
@@ -42,4 +53,81 @@ describe("[Unit] ElementAssertion.test.ts", () => {
...
+          .toThrowError(AssertionError)
+          .toHaveMessage("Expected the container to NOT contain the element");
+      });
+    });
+
+    context("when the descendant element does not contain the ancestor element", () => {
+      it("returns the assertion instance", () => {
+        const { getByTestId } = render(<ContainsTestComponent/>);
+
+        const grandparent = getByTestId("grandparent");
```
Comment (fonsiher) [inline packages/dom/test/unit/lib/ElementAssertion.test.tsx:97]: Same here with async/await

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx [hunk 3]
```diff
@@ -42,4 +53,81 @@ describe("[Unit] ElementAssertion.test.ts", () => {
...
+
+        const grandparentTest = new ElementAssertion(grandparent);
+        const parentTest = new ElementAssertion(parent);
+
+        expect(grandparentTest.toContainElement(parent));
+        expect(grandparentTest.toContainElement(child));
+        expect(grandparentTest.toContainElement(svgElement));
+        expect(parentTest.toContainElement(child));
+
+        expect(() => grandparentTest.not.toContainElement(parent))
```
Comment (fonsiher) [inline packages/dom/test/unit/lib/ElementAssertion.test.tsx:75]: I would prefer that the inverted assertion next to the normal assertion, just to be clearer of the structure of the test.

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx [hunk 4]
```diff
@@ -42,4 +53,81 @@ describe("[Unit] ElementAssertion.test.ts", () => {
...
+        const parentTest = new ElementAssertion(parent);
+        const childTest = new ElementAssertion(child);
+
+        expect(parentTest.not.toContainElement(grandparent));
+        expect(parentTest.not.toContainElement(svgElement));
+        expect(childTest.not.toContainElement(grandparent));
+        expect(childTest.not.toContainElement(parent));
+        expect(childTest.not.toContainElement(svgElement));
+
+        expect(() => parentTest.toContainElement(grandparent))
```
Comment (fonsiher) [inline packages/dom/test/unit/lib/ElementAssertion.test.tsx:111]: Same here to keep each case (regular/inverted) together. Also create the regular assertion first to maintain consistency with the previous tests.

**Review by fonsiher — COMMENTED** [review]
Awesome work on this Maty :rocket: . I just leave you a couple of non-blocking comments, which are just structural changes.

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx (hunk 1 above)
Comment (matycarolina) [reply packages/dom/test/unit/lib/ElementAssertion.test.tsx]: thanks team!

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx (hunk 3 above)
Comment (matycarolina) [reply packages/dom/test/unit/lib/ElementAssertion.test.tsx]: hi @fonsiher, thank you for the comment. I have completely updated the tests following Jose's suggestions. Please, lmk if they still need updating.

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx (hunk 4 above)
Comment (matycarolina) [reply packages/dom/test/unit/lib/ElementAssertion.test.tsx]: hi @fonsiher, same here. Please, lmk what you think. Thanks again!

**Review by ChristianSama — APPROVED** [review]
LGTM! Awesome work Maty!

**Review by fonsiher — APPROVED** [review]
Nice work Maty. LGTM !! :shipit:

File: packages/dom/test/unit/lib/fixtures/nestedElementsTestComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    <span data-testid="grandparent">
+      <span data-testid="parent">
+        <span data-testid="child">
+          <span>
+              <span>
+                  <span>
+                      <span data-testid="deep-child"></span>
+                  </span>
+              </span>
+          </span>
```
Comment (JoseLion) [inline packages/dom/test/unit/lib/fixtures/nestedElementsTestComponent.tsx:14]: 2-spaces for indentation 🙂 
```suggestion
          <span>
            <span>
              <span>
                <span data-testid="deep-child"></span>
              </span>
            </span>
          </span>
```

File: packages/dom/test/unit/lib/fixtures/withAttributesTestComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { ReactElement } from "react";
+
+export function WithAttributesTestComponent(): ReactElement {
+    return (
+      <button role="button" type="submit" className="btn primary" disabled>
```
Comment (JoseLion) [inline packages/dom/test/unit/lib/fixtures/withAttributesTestComponent.tsx:5]: Always use values on attributes (I'll add an ESLint rule later 🙂 )
```suggestion
      <button role="button" type="submit" className="btn primary" disabled={true}>
```

**Review by JoseLion — APPROVED** [review]
Looks great! I left a few minor comments. Feel free to merge once you address them 🙂

File: packages/dom/test/unit/lib/fixtures/nestedElementsTestComponent.tsx (hunk 5 above)
Comment (matycarolina) [reply packages/dom/test/unit/lib/fixtures/nestedElementsTestComponent.tsx]: Thanks @JoseLion, I have addressed this [here](https://github.com/stackbuilders/assertive-ts/pull/132/commits/229b78ed9e43e637f0b4c8f76929f913d611ae16).

File: packages/dom/test/unit/lib/fixtures/withAttributesTestComponent.tsx (hunk 6 above)
Comment (matycarolina) [reply packages/dom/test/unit/lib/fixtures/withAttributesTestComponent.tsx]: Thanks @JoseLion, I have addressed this [here](https://github.com/stackbuilders/assertive-ts/pull/132/commits/229b78ed9e43e637f0b4c8f76929f913d611ae16).

File: packages/dom/src/lib/ElementAssertion.ts [hunk 7]
```diff
@@ -30,4 +30,28 @@ export class ElementAssertion<T extends Element> extends Assertion<T> {
...
+    });
+    const invertedError = new AssertionError({
+      actual: this.actual,
+      message: "Expected the container to NOT contain the element",
+    });
+
+    return this.execute({
+      assertWhen: (
+        this.actual.contains(element)
+      ),
```
[the remark this answers is not in the document]
Comment (matycarolina) [reply packages/dom/src/lib/ElementAssertion.ts]: Thank you @JoseLion. I updated this [here](https://github.com/stackbuilders/assertive-ts/pull/132/commits/2bbe45f801e53882f2188190e1799da0e37b71f4).

File: packages/dom/src/lib/ElementAssertion.ts [hunk 8]
```diff
@@ -30,4 +30,28 @@ export class ElementAssertion<T extends Element> extends Assertion<T> {
       invertedError,
     });
   }
+
+  /**
+   * Check if a given container element contains a specified child element.
+   *
+   * @returns the assertion instance.
+   */
+  public toContainElement(element: Element): this {
```
[the remark this answers is not in the document]
Comment (matycarolina) [reply packages/dom/src/lib/ElementAssertion.ts]: Thank you @JoseLion. I updated this [here](https://github.com/stackbuilders/assertive-ts/pull/132/commits/2bbe45f801e53882f2188190e1799da0e37b71f4).

**Comment by JoseLion** [comment]
@all-contributors
please add @matycarolina for code.
please add @fonsiher for review.

**Comment by JoseLion** [comment]
@all-contributors please add @fonsiher for review.

**Pull request merged**